constructr-akka/src/main/scala/de/heikoseeberger/constructr/akka/Constructr.scala

```diff
@@ -69,9 +70,17 @@ final class Constructr private extends Actor with ActorLogging with ActorSetting
   override def receive = Actor.emptyBehavior
 
   private def active(machine: ActorRef): Receive = {
-    case Terminated(`machine`) =>
-      log.error("Terminating the system, because constructr-machine has terminated!")
-      context.system.terminate()
+    case Terminated(`machine`) => {
+      val cluster = Cluster(context.system)
+      val self = cluster.selfAddress
+      if (cluster.state.members.exists(member => member.address == self && member.status == Up)) {
```

Comment:
This check is needed to terminate the actor system in case the node is not member of the cluster yet. 
For instance a system initialization problem that prevent the node from joining the cluster. In this case a _leave_ won't do anything and the system won't be terminated.